File: auction-site-api/Data/Entities/Auction.cs
namespace auction_site_api.Data.Entities;

public class Auction
{
    public Guid Id { get; set; }
    public Guid CreatorId { get; set; }
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;
    public decimal StartPrice { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public bool IsActive { get; set; } = true;
    public ICollection<Bid> Bids { get; set; } = new List<Bid>();
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public User Creator { get; set; } = null!;
}

File: auction-site-app/src/types/auction.ts
import type { BidSummaryResponse } from '@/types/bid';

export interface AuctionCreateRequest {
    title: string;
    description: string;
    startPrice: number;
    startTime: Date;
    endTime: Date;
}

export interface AuctionListItemResponse {
    id: string;
    title: string;
    endTime: Date;
    isActive: boolean;
    currentPrice: number; // set from bids or StartPrice
}

export interface AuctionResponse {
    id: string;
    title: string;
    description: string;
    startPrice: number;
    startTime: Date;
    endTime: Date;
    isActive: boolean;
    bids: BidSummaryResponse[];
    creatorId: string;
}

export interface ClosedAuctionResponse {
    id: string;
    title: string;
    description: string;
    startPrice: number;
    startTime: Date;
    endTime: Date;
    isActive: boolean;
    winningBidAmount?: number;
    winningUserId: string;
    creatorId: string;
}

export type AuctionDetailResponse = AuctionResponse | ClosedAuctionResponse;

export interface AuctionUpdateRequest {
    title: string;
    description: string;
    endTime: Date;
    isActive: boolean;
}
